File: src/main/java/org/jhapy/notification/domain/Sms.java
package org.jhapy.notification.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * @author jHapy Lead Dev.
 * @version 1.0
 * @since 2019-07-09
 */
@Data
@EqualsAndHashCode(callSuper = true)
@Document
@ToString(callSuper = true)
public class Sms extends BaseEntity {

  private String phoneNumber;

  private String body;

  private String smsAction;

  private SmsStatusEnum smsStatus;

  private String errorMessage;

  private int nbRetry = 0;

  private String applicationName;

  private String iso3Language;
}

File: src/main/java/org/jhapy/notification/domain/Sms.kt
package org.jhapy.notification.domain

import org.springframework.data.mongodb.core.mapping.Document

/**
 * @author jHapy Lead Dev.
 * @version 1.0
 * @since 2019-07-09
 */
@Document
class Sms : BaseEntity() {
    var phoneNumber: String? = null
    var body: String? = null
    var smsAction: String? = null
    var smsStatus: SmsStatusEnum? = null
    var errorMessage: String? = null
    var nbRetry = 0
    var applicationName: String? = null
    var iso3Language: String? = null
}

enum class SmsStatusEnum {
    NOT_SENT, SENT, ERROR, RETRYING
}